homeassistant/components/met/weather.py

```diff
@@ -55,7 +55,7 @@ async def async_setup_entry(
     is_metric = hass.config.units is METRIC_SYSTEM
     if config_entry.data.get(CONF_TRACK_HOME, False):
         name = hass.config.location_name
```

Comment:
```suggestion
    if config_entry.data.get(CONF_TRACK_HOME, False):
        name = cast(str, hass.config.location_name)
```

Reply:
```
homeassistant/components/met/weather.py:57: error: Redundant cast to "str"  [redundant-cast]
```

Reply:
Oh, I would expect that this was the reason we had that TYPECHECKING

Reply:
No, It is because of the `get` on `config_entry.data`. It has type `Any` but we know it is a str.